src/android/main.cpp

```diff
@@ -54,8 +58,31 @@ extern "C"
             }
 
             if (IsVulkanReady()) {
-                VulkanDrawFrame();
+                std::vector<char*> args;
+                args.reserve(cmdArgs.size());
+                for (std::string& arg : cmdArgs) {
+                    args.push_back(const_cast<char*>(arg.c_str()));
```

Comment:
This relies on a few assumptions:
 - nativeSetCommandLineArgs is called once (otherwise pointers passed to main are now invalid)
 - this gets called after nativeSetCommandLineArgs.
 
 Are we sure that's the case? What happens when the activity gets destroyed and recreated, or just suspended (screen lock, rotation, etc).
 
 Is there a way to do it the other way around, have the native code calls an activity method returning the arguments?
 This way you'd get a copy of the arguments to pass down, and avoid this global vector which is modified and read in different locations.

Reply:
Done. This adds a lot of JNI boilerplate, but the flow makes more sense.